science/nds2-client-swig/Portfile

```diff
@@ -92,8 +106,13 @@ subport ${basename}-matlab {
   long_description          ${long_description} This package provides MATLAB \
                             bindings, modules, and scripts.
 
-  # Need GNU Classpath to get jni.h header
-  depends_build-append      port:swig-java port:gnu-classpath
+  if {[variant_isset swig3]} {
+    depends_build-append      port:swig3-java
+  }
+  java.version              1.7+
+  # Set fallback to an LTS Java version
+  java.fallback             openjdk8
```

Comment:
The newest build errors are due to using `java.version`/`java.fallback` without including the Java portgroup for the MATLAB subport.

In #8022 I thought the Java build dependency might only be needed for the Java subport rather than also the MATLAB/Octave subports, but I am not sure.

Reply:
In my latest commit, I have added the PortGroup java to the matlab subport as finding JAVA is currently needed for the installation of the matlab .m files. This causes a lint failure which I do not understand how to correct.

Reply:
The `repeats inclusion of PortGroup java` linting error is expected/can be ignored; it's a (known?) limitation of `port lint` when checking subports.